File: src/packets/PacketC2SZoneMute.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x65
 * Zone Mute
 * Mute / Unmute a Zone
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unsigned Char) Zone ID
 *      (Unsigned Char) Muting
 */
class PacketC2SZoneMute extends PacketC2S {
    parseData() {
        this._ctrllrID = this._buffer.readUInt8();
        this._zoneID = this._buffer.readUInt8();
        this._muted = (this._buffer.readUInt8() == 1);
    }

    getControllerID() {
        return this._ctrllrID;
    }

    getZoneID() {
        return this._zoneID;
    }

    getMuted() {
        return this._muted;
    }

    getID() {
        return PacketC2SZoneMute.ID;
    }
}

PacketC2SZoneMute.ID = 0x65;

module.exports = PacketC2SZoneMute;

File: src/RNetPi.Core/Packets/PacketC2SZoneMute.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x65
/// Zone Mute
/// Mute / Unmute a Zone
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (Unsigned Char) Muting
/// </summary>
public class PacketC2SZoneMute : PacketC2S
{
    public const byte ID = 0x65;

    public byte ControllerID { get; private set; }
    public byte ZoneID { get; private set; }
    public bool Muted { get; private set; }

    public PacketC2SZoneMute(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        ControllerID = Reader.ReadByte();
        ZoneID = Reader.ReadByte();
        Muted = Reader.ReadByte() == 1;
    }
}